vertx-web/src/main/java/io/vertx/ext/web/ParsedHeaderValues.java

```diff
@@ -0,0 +1,34 @@
+package io.vertx.ext.web;
+
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * A container with the request's headers that are meaningful enough to be parsed
+ * Contains:
+ * <ul>
+ * <li>Accept -> MIME header, parameters and sortable</li>
+ * <li>Accept-Charset -> Parameters and sortable</li>
+ * <li>Accept-Encoding -> Parameters and sortable</li>
+ * <li>Accept-Language -> Parameters and sortable</li>
+ * <li>Content-Type -> MIME header and parameters</li>
+ * </ul>
+ *
+ */
+public interface ParsedHeaderValues {
+  List<MIMEHeader> accept();
+  List<ParsedHeaderValue> acceptCharset();
+  List<ParsedHeaderValue> acceptEncoding();
+  List<LanguageHeader> acceptLanguage();
+  
+  MIMEHeader contentType();
+  
+  /**
+   * Given the sorted list of parsed header values the user has sent and an Iterable of acceptable values:
+   * It finds the first accepted header that matches any inside the Iterable.
+   * @param accepted The sorted list of headers to find the best one.
+   * @param in The headers to match against.
+   * @return The first header that matched, otherwise empty if none matched
+   */
+  <T extends ParsedHeaderValue> Optional<T> findBestUserAcceptedIn(List<T> accepted, Iterable<T> in);
```

Comment:
So far Maven is compiling just right. Maybe it will complain after I add those annotations. Then I'll see what has to be done. Regardless of that, these comments already called for some of the attention I'll need if it complains.